### DISCUSSION TITLE
Page taking time to load.

### DISCUSSION THREAD

**Discussion by Venkatesh-Dayani23** [Sunbird-Obsrv/Community#158] [Issues] [answered]
Hi Team,
Currently we are using 5.1.0 we noticed that the edit profile page is taking more time to load.
Below is the screenshot.Kindly need help on this and below is the error log from analytics service.

![Screenshot from 2024-04-19 17-49-31](https://github.com/Sunbird-Obsrv/Community/assets/128453924/f8f8ae74-499c-4da6-9ad6-dd003cc0115b)

_method":"POST","rid":"getDeviceProfile","title":"getDeviceProfile"}],"index":0,"data":"Ask timed out on [Actor[akka://application/user/device-profile-actor#-2101644778]] after [20000 ms]._

**Comment by sowmya-dixit** [comment]
@Venkatesh-Dayani23 Can you please share the more details from analytics service logs which will help us to understand the issue better. May be restart the analytics service and check for any errors at start up. Also share us the complete logs when the API is being hit.

**Reply by Venkatesh-Dayani23** [reply]
Hi @sowmya-dixit Below are the logs 

analytics-64677d7468-hv89x analytics 12:28:49.079 [application-akka.actor.default-dispatcher-2] INFO  org.ekstep.analytics.api - {"eid":"LOG","ets":1713788929078,"ver":"3.0","mid":"923592024CAB724774FE847FCE88D195","actor":{"id":"","type":"System"},"context":{"channel":"analytics.api","pdata":{"id":"AnalyticsAPI","ver":"3.0","pid":"getDeviceProfile","model":null},"env":"analytics","sid":null,"did":null,"cdata":null,"rollup":null},"object":null,"edata":{"type":"system","duration":0,"score":0,"rating":0.0,"level":"INFO","message":"","params":[{"status":500,"method":"POST","rid":"getDeviceProfile","title":"getDeviceProfile"}],"index":0,"data":"Get DeviceProfileAPI failed due to null","sort":null,"correlationid":null,"filters":null,"size":0},"tags":[],"@timestamp":"2024-04-22T12:28:49+00:00"}
analytics-64677d7468-hv89x analytics [error] a.a.OneForOneStrategy - null
analytics-64677d7468-hv89x analytics java.lang.NullPointerException: null
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.util.IPLocationCache$.getIpLocation(CacheUtil.scala:145)
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.service.DeviceProfileService.resolveLocation(DeviceProfileService.scala:85)
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.service.DeviceProfileService.getDeviceProfile(DeviceProfileService.scala:52)
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.service.DeviceProfileService$$anonfun$receive$1.applyOrElse(DeviceProfileService.scala:36)
analytics-64677d7468-hv89x analytics 	at akka.actor.Actor.aroundReceive(Actor.scala:539)
analytics-64677d7468-hv89x analytics 	at akka.actor.Actor.aroundReceive$(Actor.scala:537)
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.service.DeviceProfileService.aroundReceive(DeviceProfileService.scala:15)
analytics-64677d7468-hv89x analytics 	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:610)
analytics-64677d7468-hv89x analytics 	at akka.actor.ActorCell.invoke(ActorCell.scala:579)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:268)
analytics-64677d7468-hv89x analytics Restarting DeviceProfileActor: None
analytics-64677d7468-hv89x analytics Restarting DeviceProfileActor: None
analytics-64677d7468-hv89x analytics Restarting DeviceProfileActor: Some(DeviceProfileRequest(05f8d6dfc6a3ed0c821d8b93a94f2dd8,223.182.57.7))
analytics-64677d7468-hv89x analytics Restarting DeviceProfileActor: None
analytics-64677d7468-hv89x analytics Restarting DeviceProfileActor: None
analytics-64677d7468-hv89x analytics java.lang.NullPointerException
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.util.IPLocationCache$.getIpLocation(CacheUtil.scala:145)
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.service.DeviceProfileService.resolveLocation(DeviceProfileService.scala:85)
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.service.DeviceProfileService.getDeviceProfile(DeviceProfileService.scala:52)
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.service.DeviceProfileService$$anonfun$receive$1.applyOrElse(DeviceProfileService.scala:36)
analytics-64677d7468-hv89x analytics 	at akka.actor.Actor.aroundReceive(Actor.scala:539)
analytics-64677d7468-hv89x analytics 	at akka.actor.Actor.aroundReceive$(Actor.scala:537)
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.service.DeviceProfileService.aroundReceive(DeviceProfileService.scala:15)
analytics-64677d7468-hv89x analytics 	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:610)
analytics-64677d7468-hv89x analytics 	at akka.actor.ActorCell.invoke(ActorCell.scala:579)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:268)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.Mailbox.run(Mailbox.scala:229)
analytics-64677d7468-hv89x analytics Restarting DeviceProfileActor: None
analytics-64677d7468-hv89x analytics 	at akka.dispatch.Mailbox.exec(Mailbox.scala:241)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
analytics-64677d7468-hv89x analytics java.lang.NullPointerException
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.util.IPLocationCache$.getIpLocation(CacheUtil.scala:145)
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.service.DeviceProfileService.resolveLocation(DeviceProfileService.scala:85)
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.service.DeviceProfileService.getDeviceProfile(DeviceProfileService.scala:52)
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.service.DeviceProfileService$$anonfun$receive$1.applyOrElse(DeviceProfileService.scala:36)
analytics-64677d7468-hv89x analytics 	at akka.actor.Actor.aroundReceive(Actor.scala:539)
analytics-64677d7468-hv89x analytics 	at akka.actor.Actor.aroundReceive$(Actor.scala:537)
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.service.DeviceProfileService.aroundReceive(DeviceProfileService.scala:15)
analytics-64677d7468-hv89x analytics 	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:610)
analytics-64677d7468-hv89x analytics 	at akka.actor.ActorCell.invoke(ActorCell.scala:579)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:268)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.Mailbox.run(Mailbox.scala:229)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.Mailbox.exec(Mailbox.scala:241)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
analytics-64677d7468-hv89x analytics java.lang.NullPointerException
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.util.IPLocationCache$.getIpLocation(CacheUtil.scala:145)
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.service.DeviceProfileService.resolveLocation(DeviceProfileService.scala:85)
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.service.DeviceProfileService.getDeviceProfile(DeviceProfileService.scala:52)
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.service.DeviceProfileService$$anonfun$receive$1.applyOrElse(DeviceProfileService.scala:36)
analytics-64677d7468-hv89x analytics 	at akka.actor.Actor.aroundReceive(Actor.scala:539)
analytics-64677d7468-hv89x analytics 	at akka.actor.Actor.aroundReceive$(Actor.scala:537)
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.service.DeviceProfileService.aroundReceive(DeviceProfileService.scala:15)
analytics-64677d7468-hv89x analytics 	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:610)
analytics-64677d7468-hv89x analytics 	at akka.actor.ActorCell.invoke(ActorCell.scala:579)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:268)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.Mailbox.run(Mailbox.scala:229)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.Mailbox.exec(Mailbox.scala:241)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
analytics-64677d7468-hv89x analytics java.lang.NullPointerException
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.util.IPLocationCache$.getIpLocation(CacheUtil.scala:145)
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.service.DeviceProfileService.resolveLocation(DeviceProfileService.scala:85)
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.service.DeviceProfileService.getDeviceProfile(DeviceProfileService.scala:52)
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.service.DeviceProfileService$$anonfun$receive$1.applyOrElse(DeviceProfileService.scala:36)
analytics-64677d7468-hv89x analytics 	at akka.actor.Actor.aroundReceive(Actor.scala:539)
analytics-64677d7468-hv89x analytics 	at akka.actor.Actor.aroundReceive$(Actor.scala:537)
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.service.DeviceProfileService.aroundReceive(DeviceProfileService.scala:15)
analytics-64677d7468-hv89x analytics 	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:610)
analytics-64677d7468-hv89x analytics 	at akka.actor.ActorCell.invoke(ActorCell.scala:579)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:268)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.Mailbox.run(Mailbox.scala:229)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.Mailbox.exec(Mailbox.scala:241)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
analytics-64677d7468-hv89x analytics java.lang.NullPointerException
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.util.IPLocationCache$.getIpLocation(CacheUtil.scala:145)
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.service.DeviceProfileService.resolveLocation(DeviceProfileService.scala:85)
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.service.DeviceProfileService.getDeviceProfile(DeviceProfileService.scala:52)
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.service.DeviceProfileService$$anonfun$receive$1.applyOrElse(DeviceProfileService.scala:36)
analytics-64677d7468-hv89x analytics 	at akka.actor.Actor.aroundReceive(Actor.scala:539)
analytics-64677d7468-hv89x analytics 	at akka.actor.Actor.aroundReceive$(Actor.scala:537)
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.service.DeviceProfileService.aroundReceive(DeviceProfileService.scala:15)
analytics-64677d7468-hv89x analytics 	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:610)
analytics-64677d7468-hv89x analytics 	at akka.actor.ActorCell.invoke(ActorCell.scala:579)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:268)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.Mailbox.run(Mailbox.scala:229)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.Mailbox.exec(Mailbox.scala:241)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
analytics-64677d7468-hv89x analytics java.lang.NullPointerException
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.util.IPLocationCache$.getIpLocation(CacheUtil.scala:145)
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.service.DeviceProfileService.resolveLocation(DeviceProfileService.scala:85)
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.service.DeviceProfileService.getDeviceProfile(DeviceProfileService.scala:52)
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.service.DeviceProfileService$$anonfun$receive$1.applyOrElse(DeviceProfileService.scala:36)
analytics-64677d7468-hv89x analytics 	at akka.actor.Actor.aroundReceive(Actor.scala:539)
analytics-64677d7468-hv89x analytics 	at akka.actor.Actor.aroundReceive$(Actor.scala:537)
analytics-64677d7468-hv89x analytics 	at org.ekstep.analytics.api.service.DeviceProfileService.aroundReceive(DeviceProfileService.scala:15)
analytics-64677d7468-hv89x analytics 	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:610)
analytics-64677d7468-hv89x analytics 	at akka.actor.ActorCell.invoke(ActorCell.scala:579)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:268)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.Mailbox.run(Mailbox.scala:229)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.Mailbox.exec(Mailbox.scala:241)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
analytics-64677d7468-hv89x analytics 	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
analytics-64677d7468-hv89x analytics DeviceProfileService stopped successfully
analytics-64677d7468-hv89x analytics DeviceProfileService stopped successfully
analytics-64677d7468-hv89x analytics starting DeviceProfileService
analytics-64677d7468-hv89x analytics starting DeviceProfileService
analytics-64677d7468-hv89x analytics DeviceProfileService stopped successfully
analytics-64677d7468-hv89x analytics DeviceProfileService stopped successfully
analytics-64677d7468-hv89x analytics DeviceProfileService stopped successfully
analytics-64677d7468-hv89x analytics starting DeviceProfileService
analytics-64677d7468-hv89x analytics starting DeviceProfileService
analytics-64677d7468-hv89x analytics starting DeviceProfileService
analytics-64677d7468-hv89x analytics DeviceProfileService stopped successfully
analytics-64677d7468-hv89x analytics starting DeviceProfileService
analytics-64677d7468-hv89x analytics 12:28:53.658 [application-akka.actor.default-dispatcher-13] INFO  org.ekstep.analytics.api - {"eid":"LOG","ets":1713788933658,"ver":"3.0","mid":"52EC09E7428A22809F17E6CA7307FE46","actor":{"id":"","type":"System"},"context":{"channel":"analytics.api","pdata":{"id":"AnalyticsAPI","ver":"3.0","pid":"controllers.Application","model":null},"env":"analytics","sid":null,"did":null,"cdata":null,"rollup":null},"object":null,"edata":{"type":"system","duration":0,"score":0,"rating":0.0,"level":"INFO","message":"ekstep.analytics-api","params":[{"duration":44,"method":"GET","rid":"controllers.Application","size":"","status":200,"category":"","title":"controllers.Application","protocol":""},{}],"index":0,"data":null,"sort":null,"correlationid":null,"filters":null,"size":0},"tags":[],"@timestamp":"2024-04-22T12:28:53+00:00"}

**Reply by sowmya-dixit** [reply, marked as answer]
@Venkatesh-Dayani23 For analytics service we have dependency on Geo location DB setup to resolve ip. Can you confirm if this jenkins job `Provision/dev/DataPipeline/AnalyticsGeoLocationDBSetup` setup is done.

**Reply by Venkatesh-Dayani23** [reply]
No @sowmya-dixit  This job is not configured.

**Reply by sowmya-dixit** [reply]
@Venkatesh-Dayani23 Please have that job setup done, then re-deploy analytics service and let us know if you still have issues.

**Reply by Venkatesh-Dayani23** [reply]
ok @sowmya-dixit
